python/ray/data/dataset.py

```diff
@@ -339,7 +339,10 @@ def parse_filename(row: Dict[str, Any]) -> Dict[str, Any]:
                 worker pool of size ``n``, specify ``concurrency=n``. For an autoscaling
                 worker pool from ``m`` to ``n`` workers, specify ``concurrency=(m, n)``.
             ray_remote_args: Additional resource requirements to request from
-                Ray for each map worker.
+                ray for each map worker. This applies to Ray tasks and actors.
+                To request resource requirements for tasks launched by Ray actor,
+                specify ``ray_actor_task_remote_args={...}`` inside
+                ``ray_remote_args``.
```

Comment:
IIUC, the `ray_remote_args` is already used for creating the actors. but it also seems unintuitive to use a nested parameter. putting the parameter in the top tier seems better. WDYT?

Reply:
yes agree with the unintuitive behavior. right now we do not mention actor / task in top-level parameter list for `map_batches`. Seems not good to add an `ray_actor_task_remote_args`, that brings actor / task back. Another option is to use a data config to specify what's application level error to retry for actor task. WDYT?

Reply:
if currently the requirement is only for enabling application level error retry, I'd vote for adding a specific configure in DataContext. This is easier for Data users to understand.

Reply:
Updated, thanks.